gtsam/slam/OrientedPlane3Factor.cpp

```diff
@@ -19,6 +19,33 @@ void OrientedPlane3Factor::print(const string& s,
   this->noiseModel_->print("  noise model: ");
 }
 
+//***************************************************************************
+Vector OrientedPlane3Factor::evaluateError(const Pose3& pose,
+    const OrientedPlane3& plane, boost::optional<Matrix&> H1,
+    boost::optional<Matrix&> H2) const {
+  Vector err(3);
+
+  if (H1 || H2) {
+    Matrix36 predicted_H_pose;
+    Matrix33 predicted_H_plane, error_H_predicted;
+
+    OrientedPlane3 predicted_plane = plane.transform(pose, predicted_H_plane, predicted_H_pose);
```

Comment:
There is another trick here you can use to only compute derivatives when needed:

```
plane.transform(pose, H2? &predicted_H_plane : nullptr, H1 ? &predicted_H_pose : nullptr);
```

And similarly below.

Then you don’t need to branch on H1 || H2